src/portable/portable.c

```diff
@@ -420,7 +447,13 @@ static int portable_extract_by_path(
                                 return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
                                                        "Invalid item sent from child.");
 
-                        add = portable_metadata_new(iov_buffer, path, fd);
+                        /* Given recvmsg cannot be used with multiple io vectors if you don't know the size in advance,
+                         * use a marker to separate the name and the optional SELinux context. */
+                        char *selinux_label = memchr(iov_buffer, 0, n);
+                        assert(selinux_label);
```

Comment:
This is just to catch programming errors, as there should _always_ be a '\0' in the buffer - in case there was no label, there's simply another '\0' after it

Reply:
OK.